epochStart/metachain/trigger.go

```diff
@@ -214,7 +226,7 @@ func (t *trigger) Update(round uint64, nonce uint64) {
 	}
 
 	isZeroEpochEdgeCase := nonce < minimumNonceToStartEpoch
-	isNormalEpochStart := t.currentRound > t.currEpochStartRound+t.getRoundsPerEpoch()
+	isNormalEpochStart := t.currentRound > t.currEpochStartRound+t.getRoundsPerEpoch(t.epoch)
```

Comment:
I think we need some tests to check that the update to a smaller number of rounds is correctly handled in the sense that it does not set the "isNormalEpochStart" as false (check that the t.currEpochStartRound and t.epoch are updated in sync, if there is a delay between them and the Update is called in between then this will wrongly detect an invalid early change of epoch.)